language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/UsedTypeParams.scala

```diff
@@ -26,10 +31,191 @@ object UsedTypeParams {
       .foldMap(_.bifoldMap(f)(_.bifoldMap(_ foldMap (_.bifoldMap(notAGT)(f)))(f)))
   }
 
-  private def collectTypeParams(field: iface.Type): Set[String] = field match {
+  private[this] def collectTypeParams[S >: Ref.Name](field: iface.Type): Set[S] = field match {
     case iface.TypeVar(x) => Set(x)
     case iface.TypePrim(_, xs) => xs.toSet.flatMap(collectTypeParams)
     case iface.TypeCon(_, xs) => xs.toSet.flatMap(collectTypeParams)
     case iface.TypeNumeric(_) => Set.empty
   }
+
+  final case class LookupType[+RF, +VF](
+      run: iface.TypeConName => Option[(iface.DefDataType[RF, VF], LookupType[RF, VF])]
+  )
+
+  import VarianceConstraint.BaseResolution
+  import ScopedDataType.{Name => I}
```

Comment:
This is now renaming the type twice. `ScopedDataType.Name` is already an alias for `Identifier`. Can we just use `Identifier`. While I can probably guess that `I` is an identifier, I’d rather not have to guess.

Reply:
567f98a377f